src/db.c

```diff
@@ -2493,6 +2493,7 @@ void propagateDeletion(redisDb *db, robj *key, int lazy) {
 int keyIsExpired(redisDb *db, sds key, kvobj *kv) {
     /* Don't expire anything while loading. It will be done later. */
     if (server.loading) return 0;
+    if (server.allow_access_expired) return 0;
```

Comment:
if so, do we still need the check of server.allow_access_expired in expireIfNeeded()?